src/main/java/org/openhab/tools/analysis/checkstyle/KarafFeatureCheck.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle;
+
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.POM_XML_FILE_NAME;
+import static org.openhab.tools.analysis.checkstyle.api.CheckConstants.XML_EXTENSION;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.MessageFormat;
+
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheck;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+
+/**
+ * Checks if a bundle is added in a Karaf feature.xml file
+ *
+ * @author Svilen Valkanov - Initial contribution
+ */
+public class KarafFeatureCheck extends AbstractStaticCheck {
+
+    private static final String MSG_MISSING_BUNDLE_IN_FEATURE_XML = "Bundle with ID '{0}' must be added in {1}";
+    private static final String BINDING_ID_PATTERN = "mvn:{0}/{1}/{2}";
+    private static final String BINDING_FEATURE_EXPRESSION = "//features/feature/bundle[text()=\"{0}\"]";
+    private static final String POM_ARTIFACT_ID_XPATH_EXPRESSION = "//project/artifactId/text()";
+    private static final String POM_GROUP_ID_XPATH_EXPRESSION = "//project/groupId/text()";
+    private static final String POM_PARENT_GROUP_ID_XPATH_EXPRESSION = "//project/parent/groupId/text()";
+
+    private final Log logger = LogFactory.getLog(getClass());
+
+    /**
+     * Configuration property - relative path to the feature.xml file
+     */
+    private String featureXmlPath;
+
+    public KarafFeatureCheck() {
+        setFileExtensions(XML_EXTENSION);
+    }
+
+    public void setFeatureXmlPath(String featureXmlPath) {
+        this.featureXmlPath = featureXmlPath;
+    }
+
+    @Override
+    protected void processFiltered(File file, FileText fileText) throws CheckstyleException {
+        if (file.getName().equals(POM_XML_FILE_NAME)) {
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)